### DISCUSSION TITLE
Copy to OS-clipboard with multiple cursors should copy selection of all cursors, not just one

### DISCUSSION THREAD

**Discussion by EriKWDev** [helix-editor/helix#12301] [Ideas] [answered]
The action I want to perform is to re-organize everything with 'view' to be at the bottom.

```rust
let Self {
    cascades, // |  <---- cursor
    cascades_view,
    color,
    color_view,
    color_resolve,
    color_resolve_view,
    depth,
    depth_view,
    reflection,
    reflection_view,
    reflection_resolve,
    reflection_resolve_view,
    reflection_depth,
    reflection_depth_view,

    width: _,
    height: _,
    settings: _,
} = self;
```

I wanted to do the sequence `m i { s view <enter> x` to put a cursor on every `view` and then selec the whole line.

Next, I wanted to `<space> <shift> Y` to copy to os clipboard, `d` to delete, `,` to have one single cursor. Now, I wanted to navigate to just before `width` and paste everything `<space> <shift> R`, but only `cascades_view` get pasted

I think it would be more useful behaviour for the OS-clipboard to instead copy all cursors and collapse them into one text.

That, or please suggest how I do this as this is something I want to do quite often and I don't want to create a bunch of empty lines below to fit all he cursors..

**Comment by EriKWDev** [comment]
now I have to, starting at the first `_view`, repeat the action `x d p` several times until I reach the  to achieve what I want

**Comment by kirawi** [comment, marked as answer]
You need to use `:yank-joined`

**Reply by EriKWDev** [reply]
This seems like it should be default for system clipboard yanking
